### PR TITLE
PR for Amazon web assignment.

### PR DESCRIPTION
**Pull request by switchabhi**
Creating pull request for amazon web assignment.

### CODE REVIEW

**Review by poojamanna18 — COMMENTED** [review]
Comment Added for PR

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/SearchPageBL.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        this.context = Runner.getTestExecutionContext(threadId);
+        softly = Runner.getSoftAssertion(threadId);
+        String currentUserPersona = SAMPLE_TEST_CONTEXT.ME;
+        Platform currentPlatform = Runner.platform;
+    }
+
+    public ProductPageScreen selectProduct() {
+
+        ProductPageScreen productPageScreen = SearchPageScreen.get().selectProductFromSearchResultPage();
+        return productPageScreen;
```
[the remark this answers is not in the document]
Comment (switchabhi) [reply src/test/java/com/znsio/sample/e2e/businessLayer/amazon/SearchPageBL.java:32]: Added asserts & loggers

File: src/test/java/com/znsio/sample/e2e/screen/web/amazon/SearchPageScreenWeb.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        driver.waitTillElementIsPresent(bySearchResultPageMessageBarXpath);
+        return this;
+    }
+
+    @Override
+    public ProductPageScreen selectProductFromSearchResultPage() {
+        WebElement firstWidget = driver.findElement(byProductWidgetsOnSearchResultPageXpath);
+        firstWidget.click();
+        driver.switchToNextTab();
+        return this.waitForSelectProductPageScreenToLoad();
```
[the remark this answers is not in the document]
Comment (switchabhi) [reply src/test/java/com/znsio/sample/e2e/screen/web/amazon/SearchPageScreenWeb.java]: Used it for chaining. Updated it now.

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/HomePageBL.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import com.znsio.e2e.runner.Runner;
+import com.znsio.sample.e2e.businessLayer.jiomeet.InAMeetingBL;
+import com.znsio.sample.e2e.entities.SAMPLE_TEST_CONTEXT;
+import com.znsio.sample.e2e.screen.amazon.HomePageScreen;
+import com.znsio.sample.e2e.screen.amazon.SearchPageScreen;
+import org.apache.log4j.Logger;
+import org.assertj.core.api.SoftAssertions;
+
+public class HomePageBL {
+    private static final Logger LOGGER = Logger.getLogger(InAMeetingBL.class.getName());
```
[the remark this answers is not in the document]
Comment (switchabhi) [reply src/test/java/com/znsio/sample/e2e/businessLayer/amazon/HomePageBL.java]: Updated. Thanks.

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ProductPageBL.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        ProductPageScreen.get().addProductToCart();
+        //int CurrentCartCount = ProductPageScreen.get().getNumberOfProductsInCart();
+        //context.addTestState(CONTEXT_AMAZON.PRODUCT_QUANTITY,Integer.toString(CurrentCartCount));
+        //LOGGER.info(String.format("Current number of products in cart - '%d'", CurrentCartCount));
+        return this;
+    }
+
+    public ProductPageBL navigateToCart() {
+        PopupCartPageScreen.get().navigateToCartFromAddedToCartPopup();
+        return this;
```
[the remark this answers is not in the document]
Comment (switchabhi) [reply src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ProductPageBL.java:41]: Added assert & logger. Thanks.

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/HomePageBL.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        long threadId = Thread.currentThread().getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        softly = Runner.getSoftAssertion(threadId);
+        String currentUserPersona = SAMPLE_TEST_CONTEXT.ME;
+        Platform currentPlatform = Runner.platform;
+      }
+    public SearchPageBL searchForAProduct(String productName) {
+
+        SearchPageScreen homePageScreen = HomePageScreen.get().searchProduct(productName);
+        return new SearchPageBL();
```
[the remark this answers is not in the document]
Comment (switchabhi) [reply src/test/java/com/znsio/sample/e2e/businessLayer/amazon/HomePageBL.java]: Updated. Thanks.

**Pull request closed without merging**